doc-content/enterprise-only/openshift/githooks-prepare-proc.adoc

```diff
@@ -0,0 +1,41 @@
+[id='githooks-prepare-proc-{context}']
+= Preparing Git hooks 
+
+In an authoring environment you can use Git hooks to execute custom operations when the source code of a project in {CENTRAL} is changed. The typical use of Git hooks is interaction with an upstream repository. 
+
+To enable Git hooks to push commits into an upstream repository using SSH authentication, you must also provide a secret key and a known hosts file for authentication with the repository.
+
+Skip this procedure if you do not want to configure Git hooks.
+
+.Procedure
+
+. Create the Git hooks files. For instructions, see the https://git-scm.com/docs/githooks[Git hooks reference documentation].
++
+[NOTE]
+====
+A `pre-commit` script is not supported in {CENTRAL}. Use a `post-commit` script.
+====
++
+. Create a configuration map (ConfigMap) or persistent volume with the files.
+** If the Git hooks consist of one or several fixed script files, use the `oc` command to create a configuration map. For example:
++
+[subs="attributes,verbatim,macros"]
+----
+oc create configmap git-hooks --from-file=post-commit=post-commit
+----
++
+** If the Git hooks consist of long files or depend on binaries, such as executable or KJAR files, use a persistent volume. You must create a persistent volume, create a persistent volume claim and associate the volume with the claim, and transfer files to the volume. 
++
+For instructions about persistent volumes and persistent volume claims, see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.4/html/storage/index[Storage] in the {OPENSHIFT} documentation. For instructions about copying files onto a persistent volume, see https://blog.openshift.com/transferring-files-in-and-out-of-containers-in-openshift-part-3/[Transferring files in and out of containers].
++
+. If the Git hooks scripts must push commits into an upstream repository using SSH authentication, prepare a secret with the necessary files:
+.. Prepare the `id_rsa` file with a private key that matches a public key stored in the repository.
+.. Prepare the `known_hosts` file with the correct name, address, and public key for the repository.
+.. Create a secret with the two files using the `oc` command, for example:
```

Comment:
I'd mention where these files are expected to be present i.e. where the volume should be mounted
`/home/jboss/.ssh`